test/framework/klusterlet.go

```diff
@@ -0,0 +1,259 @@
+			return nil, err
+		}
+	}
+
+	// create bootstrap-hub-kubeconfig secret
+	secret := bootstrapHubKubeConfigSecret.DeepCopy()
+	if _, err := spoke.KubeClient.CoreV1().Secrets(agentNamespace).Get(context.TODO(), secret.Name, metav1.GetOptions{}); err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.Errorf("failed to get secret %v in ns %v. %v", secret.Name, agentNamespace, err)
+			return nil, err
+		}
+		if _, err = spoke.KubeClient.CoreV1().Secrets(agentNamespace).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+			klog.Errorf("failed to create secret %v in ns %v. %v", secret, agentNamespace, err)
+			return nil, err
+		}
+	}
+
+	if helpers.IsHosted(mode) {
+		// create external-managed-kubeconfig, will use the same cluster to simulate the Hosted mode.
+		secret.Namespace = agentNamespace
+		secret.Name = helpers.ExternalManagedKubeConfig
+		if _, err := spoke.KubeClient.CoreV1().Secrets(agentNamespace).Get(context.TODO(), secret.Name, metav1.GetOptions{}); err != nil {
+			if !apierrors.IsNotFound(err) {
+				klog.Errorf("failed to get secret %v in ns %v. %v", secret.Name, agentNamespace, err)
+				return nil, err
+			}
+			if _, err = spoke.KubeClient.CoreV1().Secrets(agentNamespace).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+				klog.Errorf("failed to create secret %v in ns %v. %v", secret, agentNamespace, err)
+				return nil, err
+			}
+		}
+	}
+
+	// create klusterlet CR
+	realKlusterlet, err := spoke.OperatorClient.OperatorV1().Klusterlets().Create(context.TODO(),
+		klusterlet, metav1.CreateOptions{})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("failed to create klusterlet %v . %v", klusterlet.Name, err)
+		return nil, err
+	}
+
+	return realKlusterlet, nil
+}
+
+func CreatePureHostedKlusterlet(spoke *Spoke, name, clusterName string) (*operatorapiv1.Klusterlet, error) {
+	if name == "" {
+		return nil, fmt.Errorf("the name should not be null")
+	}
+
+	var klusterlet = &operatorapiv1.Klusterlet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: operatorapiv1.KlusterletSpec{
+			RegistrationImagePullSpec: "quay.io/open-cluster-management/registration:latest",
+			WorkImagePullSpec:         "quay.io/open-cluster-management/work:latest",
+			ExternalServerURLs: []operatorapiv1.ServerURL{
+				{
+					URL: "https://localhost",
+				},
+			},
+			ClusterName: clusterName,
+			DeployOption: operatorapiv1.KlusterletDeployOption{
+				Mode: operatorapiv1.InstallModeHosted,
+			},
+		},
+	}
+
+	// create klusterlet CR
+	realKlusterlet, err := spoke.OperatorClient.OperatorV1().Klusterlets().Create(context.TODO(),
+		klusterlet, metav1.CreateOptions{})
+	if err != nil {
+		klog.Errorf("failed to create klusterlet %v . %v", klusterlet.Name, err)
+		return nil, err
+	}
+
+	return realKlusterlet, nil
+}
+
+func CheckKlusterletStatus(spoke *Spoke, klusterletName, condType, reason string, status metav1.ConditionStatus) error {
```

Comment:
spoke already has the klusterName set in it, shouldn't this as a func under spoke struc?

Reply:
The spoke has the name of `KlusterletOperator` instead of `Klusterlet`. The same reason in https://github.com/open-cluster-management-io/ocm/pull/565#discussion_r1675258355